yt/data_objects/data_containers.py

```diff
@@ -793,9 +795,18 @@ def create_firefly_object(
         ## for safety, in case someone passes a float just cast it
         default_decimation_factor = int(default_decimation_factor)
 
+        if JSONdir is not None:
+            issue_deprecation_warning(
+                "The 'JSONdir' keyword argument is a deprecated alias for 'datadir'."
+                "Please use 'datadir' directly.",
+                since="4.1",
+                removal="4.2",
+            )
+            datadir = JSONdir
+
```

Comment:
an error should be raised (`TypeError`) if `datadir` is still `None` at this point.

Reply:
`None` is an allowed (and is the default) value for `datadir` though. From the Firefly doc/code, a value of `none` defaults to `$HOME/<file_prefix>` (and `file_prefix` defaults to `Data`).

Reply:
Yes that would make sense indeed ! While we're at it, we should probably also document where additional `**kwargs` are being passed to, since that's also supported.